docs/how-we-work.md

```diff
@@ -131,19 +131,28 @@ The github package ships as a bundled part of Atom, which affects the way that o
+6. Create a [QA issue](https://github.com/atom/github/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3Aquality) in the atom/github repository. Its title should be "_prerelease version_ QA Review" and it should have the "quality" label applied. Populate the issue body with a checklist containing the pull requests that were included in this release; these should be the ones in the "Merged" column of the project board. Omit pull requests that don't have verification steps (like renames, refactoring, adding tests or metrics, or dependency upgrades, for example).
 7. Use your `atom-dev` build to verify each and check it off the list.
-  * :boom: _If verification fails,_ note the failure in an issue comment. Close the issue. Correct the failure with more work in the current sprint board, then begin again at (1).
+  * :boom: _If verification fails,_ 
+    1. Note the failure in an issue comment. Close the issue. 
+    1. Correct the failure with more work in the current sprint board. Make changes on master branch.
+    1. Cherry changes from master to release branch.
+    1. Begin again by cutting a new pre-release and proceeding through the above steps once again.
   * :white_check_mark: _Otherwise,_ comment in and close the issue, then continue.
 8. _In your atom/github repository:_ run `apm publish minor` to publish the next minor version.
+  * :boom: _If publishing fails,_ before trying to publish again
+    1. Check if a release commit was created (`git log`). If one exists, remove it from the commit history (`git reset --hard <sha-prior-to-commit>`).
+    1. Check if a release tag was created (`git tag`). If one exists, delete it (`git tag -d 0.${MINOR}.0`).
+    1. Address the problem that interfered with publishing.
+    1. Try to publish again with `apm publish minor`.
 9. _In your atom/github repository:_ create a release branch for this minor version with `git checkout -b 0.${MINOR}-releases`. Push it to atom/github.
-9. _In your atom/atom repository:_ update the version of the `"github"` entry beneath `packageDependencies` in `package.json`  to match the published minor version. Run `script/build` to update `package-lock.json` files. Commit and push these changes.
+9. _In your atom/atom repository:_ checkout a new branch (`git checkout -b bump-github-${VERSION}`), update the version of the `"github"` entry beneath `packageDependencies` in `package.json`  to match the published minor version. Run `script/build` to update `package-lock.json` files. Commit and push these changes.
```

Comment:
Sidenote: you can get away with just running `script/bootstrap` instead of `script/build`. But `script/build` is safer because it actually runs through the final snapshotting and everything, so let's leave it like this.